### PR TITLE
Update first-class-structs.md

### PR DESCRIPTION
**Pull request by CarolEidt**

### CODE REVIEW

**Comment by CarolEidt** [comment]
@dotnet/jit-contrib PTAL
I updated the doc to reflect current status.
I've eliminated the old and outdated IR listings, and have fixed some old issues to refer to the runtime repo.

File: docs/design/coreclr/jit/first-class-structs.md [hunk 1]
```diff
@@ -213,16 +225,21 @@ This would be done in multiple phases:
       for a general idea of the kinds of VM changes that may be required.
   * Defer retyping of struct return types (`Compiler::impFixupStructReturnType()` and
     `Compiler::impFixupCallStructReturn()`)
-    * This is probably the "right" way to fix [#13355](https://github.com/dotnet/runtime/issues/13355).
+    * This has largely (fully?) been addressed with [#37745 Disable JitDoOldStructRetyping by default](https://github.com/dotnet/runtime/pull/37745).
```
Comment (sandreenko) [inline docs/design/coreclr/jit/first-class-structs.md:228]: I would say it was fully addressed.

File: docs/design/coreclr/jit/first-class-structs.md [hunk 2]
```diff
@@ -213,16 +225,21 @@ This would be done in multiple phases:
...
   * Next, eliminate the "pessimizations".
     * For cases where `GT_LCL_FLD` is currently used to "retype" the struct, change it to use *either*
       `GT_LCL_FLD`, if it is already address-taken, or to use a `GT_BITCAST` otherwise.
```
Comment (sandreenko) [inline docs/design/coreclr/jit/first-class-structs.md:231]: I would add https://github.com/dotnet/runtime/issues/11413 to this list.

**Review by sandreenko — APPROVED** [review]
LGTM, thank you!

File: docs/design/coreclr/jit/first-class-structs.md (hunk 2 above)
Comment (CarolEidt) [reply docs/design/coreclr/jit/first-class-structs.md:233]: Done, thanks

**Pull request merged**